core/shared/src/main/scala/scalaz/zio/IO.scala

```diff
@@ -425,32 +425,28 @@ sealed abstract class IO[+E, +A] { self =>
    * completes, or until the first failure.
    */
   final def repeat[B](schedule: Schedule[A, B]): IO[E, B] =
-    repeatOrElse[E, B](schedule, (e, _) => IO.fail(e))
+    repeatOrElse[B, E](schedule, (e, _) => IO.fail(e))
```

Comment:
Ooh subtle `-B` vs `+E` 😃

Reply:
Good catch, the other order was better, I think.